src/main/scala/cromwell/engine/WorkflowManager.scala

```diff
@@ -0,0 +1,83 @@
+package cromwell.engine
+
+import java.util.UUID
+
+import akka.actor.{Actor, ActorSystem, ActorRef}
+import akka.pattern.pipe
+import scala.concurrent.ExecutionContext.Implicits.global
+import cromwell.binding.{WdlBinding, WdlInputs, Wdl}
+import cromwell.engine.WorkflowActor.Start
+import cromwell.engine.WorkflowManagerActor.{SubmitWorkflow, WorkflowStatus}
+import cromwell.engine.backend.Backend
+import cromwell.engine.backend.local.LocalBackend
+
+import scala.concurrent.Future
+
+/**
+ * Abstract notion of a workflow manager. Mainly exists to allow for actor-system-free unit testing of basic concepts
+ */
+trait WorkflowManager {
+  type Workflow
+
+  case class ManagedWorkflow(id: WorkflowId, workflow: Workflow)
+
+  def backend: Backend
+
+  def generateWorkflow(wdl: Wdl, inputs: WdlInputs): Workflow
+  def workflowStatus(id: WorkflowId): Option[WorkflowState]
+
+  private var managedWorkflows = Map.empty[WorkflowId, Workflow]
+
+  def submitWorkflow(wdl: Wdl, inputs: WdlInputs): Option[ManagedWorkflow] = {
+    val managedWorkflow = ManagedWorkflow(UUID.randomUUID(), generateWorkflow(wdl, inputs))
+    if (managedWorkflows contains managedWorkflow.id) None
+    else {
+      managedWorkflows += managedWorkflow.id -> managedWorkflow.workflow
+      Option(managedWorkflow)
+    }
+  }
+
+  def workflowById(id: WorkflowId): Option[Workflow] = managedWorkflows.get(id)
+}
+
+/**
+ * A WorkflowManager using WorkflowActors as the workflow handler
+ */
+trait ActorWorkflowManager extends WorkflowManager {
+  override type Workflow = ActorRef // TODO: In a world where Akka Typed is no longer experimental switch to that
```

Comment:
The problem is that ActorRef can be an ActorRef of any Actor and there's no way around that at the moment. A modification to Akka expected later this year would allow us to require at compile time that it's an ActorRef[Workflow] (or whatever the syntax). 

At the moment somewhere you gotta pay the piper that you have an untyped ActorRef rolling around.